usage/usageplot.py

```diff
@@ -85,23 +93,107 @@ def make_plot(title, times, nodes, name):
     nodes : Numpy array of `int`
         array of the node values from the data_file
     name : `str`
-        name of the plot png file 
-    """ 
+        name of the plot png file
+    jobs : `list` of `str`
+        list of the job names from SLURM at a given time step
+    color : `bool`
+        if the plot will be color-coded by job name
+    """
     plt.figure(figsize=(8, 8))
     plt.plot(times, nodes, 'b', drawstyle="steps-post")
     plt.ylim(0, 50)
     plt.xlim(xmin=0)
     plt.grid(linestyle=':')
-    plt.fill_between(times, 0, nodes, step="post", facecolor='b', alpha=0.25)
+    
+    if color:
+        start_end = get_first_last(jobs)
+        colors = {'singleFrame': 'c', 'mosaic': 'xkcd:yellow', 'coadd': 'g',
+                  'multiband': 'b', 'unknown': 'r'}
+        hatch = {'singleFrame': '///', 'mosaic': '\\\\', 'coadd': '...',
+                 'multiband': 'x', 'unknown': '**'}
+
+        for key in start_end.keys():
+            for val_tup in start_end[key]:
+                plt.fill_between(times[val_tup[0]:val_tup[1]+2], 0,
+                                 nodes[val_tup[0]:val_tup[1]+2], hatch=hatch[key],
+                                 step="post", facecolor=colors[key], alpha=0.5,
+                                 label=key if key not in
+                                 plt.gca().get_legend_handles_labels()[1] else '')
+        plt.legend(title='Code Name')
+    
+    else:
+        plt.fill_between(times, 0, nodes, step="post", facecolor='b', alpha=0.25)
+
     plt.xlabel(r'time [h]', fontsize=16)
     plt.ylabel(r'$\ N_{node} $', fontsize=16)
     plt.tick_params(axis='both', which='major', labelsize=14)
     plt.title(title, fontsize=18)
     plt.savefig(name + ".png")
 
 
+def get_first_last(jobs):
+    """Gets the first and the last index of the code names of the same type,
+    (i.e.: will return the first and last index of the singleFrameDriver jobs)
+    which will allow for color-coding in make-plot.
+
+    Parameters
+    ----------
+    jobs : `list` of `str`
+        list of the job names from SLURM at a given time step
+
+     Returns
+    -------
+    start_end : `dict` of `list` of `tuple`
+        dictionary containing code names and the start and stop indexes
+        of the code runs.
+    """
+    names = dict()
```

Comment:
Please,, put this line *after* the comment. Now, at the first glance, it seems the code below the comment is not necessarily related to this particular dictionary.